packages/@lwc/engine/src/framework/def.ts

```diff
@@ -236,15 +258,19 @@ export function isComponentConstructor(ctor: any): ctor is ComponentConstructor
  * component metadata. This API is subject to change or being removed.
  */
 export function getComponentDef(Ctor: any, subclassComponentName?: string): ComponentDef {
+    if (!isComponentConstructor(Ctor)) {
+        throw new TypeError(
+            `${Ctor} is not a valid component, or does not extends LightningElement from "lwc". You probably forgot to add the extend clause on the class declaration.`
+        );
+    }
+
+    if (isCircularModuleDependency(Ctor)) {
+        Ctor = resolveCircularModuleDependency(Ctor);
+    }
+
     let def = CtorToDefMap.get(Ctor);
```

Comment:
This WeakMap lookup offered caching to resolve circular reference once per Ctor. The `getComponentDef` routine is on the critical path because it runs every time a vm is created. Having the WeakMap lookup is still faster IMO.